### ISSUE TITLE
Docs: More practical example for createAction without duplication?

### ISSUE THREAD

**Issue by Izhaki** [reduxjs/redux-toolkit#3454]
At the moment, the [docs](https://redux-toolkit.js.org/usage/usage-with-typescript#createaction) show this example for action creation that support union types with no duplication:

```ts
function withPayloadType<T>() {
  return (t: T) => ({ payload: t })
}
createAction('test', withPayloadType<string>())
```

This is really nice if one wants to understand how it works. But ultimately, if you do this time and again you might end up writing:

```ts
const createActionEx = <T>(type: string) =>
  createAction(type, (t: T) => ({ payload: t }));

createActionEx<string>('test')
``` 

Maybe worth considering adding this to the docs (possibly replacing the current code?).

**Comment by markerikson** [comment]
Out of curiosity, what's your use case for actually calling `createAction` in the first place?

Our assumption is that most users will never even need to call it, because `createSlice` and `createAsyncThunk` already do that internally.

**Comment by Izhaki** [comment]
Glad you asked.

I've been using redux for 7 years now, and in many cases I've used it as a reactive framework + state manager. The reactive framework part = middlewares.

One example was to create a **game server simulator** for testing purposes. With cucumber dispatching into redux messages that would otherwise come from the server.

When used this way, many actions (if not most, depending on the problem at hand), do not end up as state-mutators.

To give current example, in a graphical editing framework you would have this flow:

- The UI dispatches `PointerEvent` action.
- Which the Pointer middleware converts to a `Drag` action.
- Which the Move Tool middleware converts to a `Move` action.
- Which the Edit Policy middleware converts to a `Resize` action
- Which mutates the state.

---

Anyhow, the example above does not work (I swear it was before I filed this "issue" - no idea!). So closing.
